qiskit/circuit/library/standard_gates/multi_control_rotation_gates.py

```diff
@@ -0,0 +1,239 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Multiple-Controlled U3 gate. Not using ancillary qubits.
```

Comment:
this is mcrx, mcry, not u2

Reply:
u3 -- actually i don't know what this gate is.

Reply:
@Cryoris  can you look into this. It says MCU3 but produces MCRX MCRY MCRZ.
Also does this algorithm give you the same output when theta=pi as the one for MCX?

Reply:
The method there use a gray code to control the u3 gate on an arbitrary number of controls. Then this multi-controlled u3 is used to define multi-controlled rx, ry and rz rotations. If used as `MCU3(pi, 0, pi)` this should be equivalent to `MCX` with the gray code mode. 

But I don't think these gates actually add value, our `add_control` implementation uses the same technique. Only for `mcry` there is an alternate implementation using ancillas which make sense to keep. Should we put these in `library/generalized_gates` to have a complete set of multi-controlled gates? (using the default `add_control`)

Reply:
just make an issue to move it later so this thing with 106 files can be merged.

Reply:
Added it to #4316, where this is already mentioned